src/credentials/CHIPCert.cpp

```diff
@@ -1205,12 +1199,11 @@ CHIP_ERROR ConvertIntegerDERToRaw(ByteSpan derInt, uint8_t * rawInt, const uint1
 
 CHIP_ERROR ConvertECDSASignatureRawToDER(P256ECDSASignatureSpan rawSig, MutableByteSpan & derSig)
 {
-    ASN1Writer writer;
+    VerifyOrReturnError(derSig.size() >= rawSig.size() + kEmitDerIntegerOverhead, CHIP_ERROR_BUFFER_TOO_SMALL);
```

Comment:
Thanks, looks like I got some constants mixed up here. The comment for the method says the buffer needs to be at least `kMax_ECDSA_Signature_Length_Der` so that's what we should check. Would be good to sanity check that the output did indeed fit, but that doesn't really help the caller size the buffer correctly in the first place.

Reply:
Fixed the size check. A sanity check already exists in `MutableSpan::reduce_size`